transport/shadowsocks/stream_test.go

```diff
@@ -185,59 +185,35 @@ func TestLazyWriteFlush(t *testing.T) {
 	writer := NewWriter(buf, key)
 	header := []byte{1, 2, 3, 4}
 	n, err := writer.LazyWrite(header)
-	if n != len(header) {
-		t.Errorf("Wrong write size: %d", n)
-	}
-	if err != nil {
-		t.Errorf("LazyWrite failed: %v", err)
-	}
-	if buf.Len() != 0 {
-		t.Errorf("LazyWrite isn't lazy: %v", buf.Bytes())
-	}
-	if err = writer.Flush(); err != nil {
-		t.Errorf("Flush failed: %v", err)
-	}
+	require.Nilf(t, err, "LazyWrite failed: %v", err)
```

Comment:
This is clean and tidy, any thoughts on using the `require` across all of our repos besides just tests?

Reply:
How would we do that?
`require` is intended to be used in test. It receives a `*testing.T` in order to propagate the error. We could use `assert`, which returns true or false based on the assertion, but it also needs a `*testing.T` and we would still need an if block.